packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -438,6 +451,23 @@ export abstract class BaseDataSource extends Construct implements IGrantable {
 
 }
 
+/**
+ * Properties for an AppSync dummy datasource
+ */
+export interface NoneDataSourceProps extends BaseDataSourceProps {
```

Comment:
Ideally this datasource wouldn't allow you to set serviceRole at all.

My hacky workaround to this in https://github.com/aws/aws-cdk/issues/6836#issuecomment-610722978 was to do:

```typescript
export interface NoneDataSourceProps extends BaseDataSourceProps {
  readonly serviceRole?: undefined
}
```

Reply:
One way to implement this would be to remove serviceRole from BaseDataSourceProps, and instead add it to a new BackedDataSourceProps interface that others can inherit from.

Reply:
Sure, we'll try to do this early next week.

Reply:
I have extracted `serviceRole` to `BackedDataSourceProps`. Please have a look at https://github.com/aws/aws-cdk/pull/7145/commits/154390d9101bacac2dcec68e5fe7e90a7c51d14f

Reply:
The question still remains if `NoneDateSourceProps` should exist at all, or if `NoneDateSource` should just accept `BaseDataSourceProps`?

Reply:
Let's keep NoneDataSourceProps for now.